extract_timeseries_tar.py

```diff
@@ -148,37 +285,51 @@ def get_parser():
     data_path = args.input_dir
     atlas_path = args.atlas_path
     dataset_name = args.dataset_name
+    download_only = args.download_only
     output_root_dir = args.output_dir
 
-    layout = bids.BIDSLayout(data_path, config=['bids', 'derivatives'])
-    subject_list = layout.get(return_type='id', target='subject')
+    if download_only:
+        print("Download only.")
+        download_atlases()
+    else:
+        template = 'MNI152NLin2009cAsym'
+        resolution = "02"
 
-    for atlas_name in ATLAS_METADATA.keys():
-        print("-- {} --".format(atlas_name))
-        dataset_title = f"dataset-{dataset_name}_atlas-{atlas_name}"
-        output_dir = os.path.join(output_root_dir, dataset_title)
-        os.makedirs(output_dir, exist_ok=True)
-        for subject in subject_list:
-            print(f"sub-{subject}")
-            # TODO: loop through all
-            # Note from AB
-            # if multiple run, use run 2
-            # if multiple session, use ses 1
-            fmri = layout.get(return_type='type', subject=subject, space='MNI152NLin2009cAsym',
-                              desc='preproc', suffix='bold', extension='nii.gz')
-            for ii in range(len(fmri)):
-                file_entitiles = fmri[ii].entities
-                timeseries_root_dir = create_timeseries_root_dir(
-                    file_entitiles, output_dir)
-                for dimension in ATLAS_METADATA[atlas_name]['dimensions']:
-                    for resolution in ATLAS_METADATA[atlas_name]['resolutions']:
-                        print(f"\t dim{dimension} - res0{resolution}")
+        layout = bids.BIDSLayout(data_path, config=['bids', 'derivatives'])
+        subject_list = layout.get(return_type='id', target='subject')
+
+        for atlas_name in ATLAS_METADATA.keys():
+            print("-- {} --".format(atlas_name))
+            dataset_title = f"dataset-{dataset_name}_atlas-{atlas_name}"
+            output_dir = os.path.join(output_root_dir, dataset_title)
+            os.makedirs(output_dir, exist_ok=True)
+            update_templateflow_path(atlas_name, atlas_path)
+            for subject in subject_list:
+                print(f"sub-{subject}")
+                # TODO: loop through all
+                # Note from AB
+                # if multiple run, use run 2
+                # if multiple session, use ses 1
```

Comment:
Can be removed as this is specific for CCNA?
```suggestion
```

Reply:
Not sure why it is still there indeed.
A more elegant solution would be:
```
fmri = layout.get(return_type='type', subject=subject, space=template,
                                  desc='preproc', suffix='bold', extension='nii.gz')
if isinstance(fmri, list):
    fmri = fmri[0]
```